cmd/alerts/list_alerts.go

```diff
@@ -0,0 +1,124 @@
+package alerts
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+)
+
+var levelCmd string
+
+type alertCmd struct {
+	clusterID  string
+	alertLevel string
+}
+
+type Labels struct {
+	Alertname string `json:"alertname"`
+	Severity  string `json:"severity"`
+}
+
+type Status struct {
+	State string `json:"state"`
+}
+
+type Annotations struct {
+	Summary string `json:"summary"`
+}
+
+type Alert struct {
+	Labels      Labels      `json:"labels"`
+	Status      Status      `json:"status"`
+	Annotations Annotations `json:"annotations"`
+}
+
+// osdctl alerts list ${CLUSTERID} --level [warning, critical, firing, pending, all] --active bool
+func NewCmdListAlerts() *cobra.Command {
+	alertCmd := &alertCmd{}
+	newCmd := &cobra.Command{
+		Use:               "list <cluster-id>",
+		Short:             "list alerts",
+		Long:              `Checks the alerts for the cluster`,
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			alertCmd.clusterID = args[0]
+			ListAlerts(alertCmd)
+		},
+	}
+
+	newCmd.Flags().StringVarP(&alertCmd.alertLevel, "level", "l", "", "Alert level [warning, critical, firing, pending, all]")
+	return newCmd
+}
+
+func alertLevel(level string) string {
+	switch level {
+	case "warning":
+		levelCmd = "warning"
+	case "critical":
+		levelCmd = "critical"
+	case "firing":
+		levelCmd = "firing"
+	case "pending":
+		levelCmd = "pending"
+	case "all":
+		levelCmd = "all"
+	default:
+		log.Fatalf("Invalid alert level: %s\n", level)
+		return ""
+	}
+	return levelCmd
+}
+
+func ListAlerts(cmd *alertCmd) {
+	var alerts []Alert
+
+	defer func() {
+		if err := recover(); err != nil {
+			log.Fatal("error : ", err)
+		}
+	}()
+
+	clusterID := cmd.clusterID
+	levelcmd := cmd.alertLevel
```

Comment:
nitpicky - I recommend declaring variables no sooner than they're needed. For example, the instantiation of `levelcmd` here could be moved down to line 99/100. It doesn't make as much of a difference with static values like this, but it's a good habit to get into for assignments that could have side effects.

Reply:
@AlexVulaj I don't get it the full context here, `clusterID` and `levelcmd` as argument in next line, if we declare in line 99/100 that would be calling function before declaration and code will through warning ?

Reply:
As it's related to `levelcmd` - I dislike that this function as essentially the same variable with 2 different capitalizations being the only difference:
- `levelCmd`
- `levelcmd`
These also seem to be used to store the same thing (the level of alerts to list), so please only use a single variable for it.
I would also recommend calling it `alertLevel` instead maybe, as the `cmd` prefix made me think it would store some shell-command to execute.